lib/src/repo_path.rs

```diff
@@ -858,4 +923,29 @@ mod tests {
             Ok(repo_path("dir/file"))
         );
     }
+
+    #[test]
+    fn test_format_copied_path() {
+        let ui = RepoPathUiConverter::Fs {
+            cwd: PathBuf::from("."),
+            base: PathBuf::from("."),
+        };
+
+        let format = |before, after| {
+            ui.format_copied_path(repo_path(before), repo_path(after))
+                .replace("\\", "/")
+        };
+
+        assert_eq!(format("one/two/three", "one/two/three"), "one/two/three");
+        assert_eq!(format("two/three", "four/three"), "{two => four}/three");
+        assert_eq!(
+            format("one/two/three", "one/four/three"),
+            "one/{two => four}/three"
+        );
+        assert_eq!(format("one/two/three", "one/three"), "one/{two => }/three");
```

Comment:
Could you add tests like the following too?
```
        assert_eq!(format("one/two", "one/two/three"), "one/{two => two/three}");
        assert_eq!(format("one/two", "zero/one/two"), "{one/ => zero/one/}two");
        assert_eq!(format("one/two/three", "one/two"), "one/{two/three => three}");
        assert_eq!(format("zero/one/two", "one/two"), "{zero/one/ => one/}two");
        assert_eq!(format("one/two", "one/two/three/one/two"), "one/{two => two/three/one/two}");
```

I think that's how I would expect the result to look but I'm not sure. You can check what git does if you feel like it.

Reply:
done